### ISSUE TITLE
Appium inspector stuck with loading spinner forever: iOS 13 Real Device

### ISSUE THREAD

**Issue by venkatraman91g** [appium/appium#13256]
## The problem
When tried to inspect Settings app in iOS 13 using appium inspector, the app got launched but the inspector window keeps on showing loading spinner indefinitely. 

## Environment

* Appium version (or git revision) that exhibits the issue: 1.13.0
* Last Appium version that did not exhibit the issue (if applicable): 1.12.0
* Desktop OS/version used to run Appium: Mac OS 10.14.5
* Node.js version (unless using Appium.app|exe): 
* Npm or Yarn package manager:
* Mobile platform/version under test: iOS 13.1 Beta 2
* Real device or emulator/simulator: Real Device
* Appium CLI or Appium.app|exe:

## Link to Appium logs

https://gist.github.com/venkatraman91g/a3e8cbfe72a4a319ad22c3f4c0f20ce3

## Code To Reproduce Issue [ Good To Have ]

Tried inspecting app using Appium Desktop

**Comment by KazuCocoa** [comment]
You should run with `appium@beta` for iOS 13+.
https://github.com/appium/appium-desktop/#connect-to-local-appium-server is the hint to connect to local appium from the desktop.

**Comment by venkatraman91g** [comment]
Yes. That worked. Thanks a lot.

**Comment by awhisler-nutrien** [comment]
This does **not** work for me! Help!

**Comment by amitsahu86** [comment]
Appium Inspector is not working in my case.

Scenario 1:
- Launch appium server from terminal
- Open Appium Desktop
- Select New Session Window in menu
- Tab to Custom Server
- Remote host: localhost, Remote port: 4723, Remote path: /wd/hub
- Added my capabilities
- Start session

Issues: The error message show - The URL '/wb/hub/session' did not map to a valid resource
<img width="1076" alt="Screenshot 2019-10-04 at 3 08 42 PM" src="https://user-images.githubusercontent.com/8972470/66198439-0f821680-e6ba-11e9-934b-50e60299ffdd.png">

Log on appium server on terminal:
[debug] [GENERIC] Calling AppiumDriver.getSessions() with args: []
[debug] [GENERIC] Responding to client with driver.getSessions() result: []
[HTTP] <-- GET /wd/hub/sessions 200 3 ms - 40
[HTTP] 
[HTTP] --> POST /wb/hub/session
[HTTP] {"desiredCapabilities":{"app":"path-to-app","bundledId":"com.xyz","deviceName":"iPhone 8","deviceUDID":"C70FBC5C-1234-48E9-1234-9C24EF043E9F","platformName":"iOS","platformVersion":"13.0","simpleIsVisibleCheck":"true","useJSONSource":"true","newCommandTimeout":0,"connectHardwareKeyboard":true}}
[debug] [HTTP] No route found. Setting content type to 'text/plain'
[HTTP] <-- POST /wb/hub/session 404 1 ms - 57
[HTTP] 
[HTTP] --> DELETE /wb/hub/session
[HTTP] {}
[debug] [HTTP] No route found. Setting content type to 'text/plain'
[HTTP] <-- DELETE /wb/hub/session 404 1 ms - 57
[HTTP]

Scenario 2:
- Open Appium Desktop
- Entered Host - 0.0.0.0 and Port - 4724
- Start Appium Server 1.13
- Click on Appium Inspector icon (Search icon)
- Tab to Custom Server
- Remote host: 0.0.0.0, Remote port: 4724, Remote path: /wd/hub
- Added my capabilities
- Start session

Issue:  Screen display Infinite loading

**Comment by bokiperic** [comment]
Is there a solution to this? Since I updated XCode to 11.1 and tried to start Appium Inspector on iOS Simulator (iOS 13,  12.4,  12.1) I get the "endless spinning" (or "infinite loading") while in Appium server I get:
` [HTTP] <-- GET /wd/hub/session/2692fffa-068a-4176-84a3-59b41b38d4ed/window/current/size 500`
On iOS simulator the app is started, but for Appium Inspector endless spinning is present. Appium Server version is 1.14.0 (1.14.0.20190705.4)

I looked at the referenced issue, but it seems that both, this one and that one, are closed w/o resolving the problem.

**Comment by ameadors** [comment]
Inspector is spinning when loading against iOS projects, does not time out or complete.
Running Xcode 11 Version 11.1 (11A1027) and Appium Version 1.15.0-1 (1.15.0-1.20191004.1)

**Comment by ameadors** [comment]
just tried Appium desktop build Version 1.15.1 (1.15.1.20191013.2), still no fix for the broken/spinning inspector with iOS

**Comment by KazuCocoa** [comment]
Can you check the cause is appium server or Appium Desktop?
If you can get `page source`, it is helpful to create an issue on https://github.com/appium/appium-desktop to track it correctly.

**Comment by vzenzo** [comment]
Is there any news on this issue? This is still happening to me

**Comment by u546012** [comment]
> just tried Appium desktop build Version 1.15.1 (1.15.1.20191013.2), still no fix for the broken/spinning inspector with iOS

is this issue fixed, I am also facing this issue

**Comment by nitinsurange** [comment]
@u546012 Please use appium desktop beta version 1.17.0 and it will work for you.

Download from here https://github.com/appium/appium-desktop/releases/tag/v1.17.0

I'm using xcode 11.4 and deployment ios target 13.4 .

Let me know if you  still face problem.

**Comment by laxminarayanjena-gl** [comment]
@nitinsurange i am using xcode 11.4.1 and appium gui desktop 1.17 still i am getting spiining issues
https://github.com/appium/appium-desktop/issues/1358

**Comment by rbradhika530** [comment]
I am facing spinning issue with iOS 15.4.1, while using appium inspector

**Comment by MrunalCTS** [comment]
I am facing spiing issues with appium inspector 1.19.1, appium 1.22.3 and iOS versions from 13 to 15.6.1. Android devices load fine. Is there any solution/checlist on this please?

**Comment by OleksiiRudyi** [comment]
Try the solution in this thread. It helped me.
https://github.com/appium/appium/issues/19382

**Comment by Shilpa-New** [comment]
I have faced similar issue recently.

Two different issues: 
1) While running automations script got error:
"An unknown server-side error occurred while processing the command. Could not proxy command to remote server. Original error: Error: socket hang up Solution:"
2) When we connected with Appium Inspector, loader keeps loading for infinite, Does not open App screen.

**Common Solution:**
Execute on command prompt- 
    1. adb uninstall io.appium.uiautomator2.server
    2. adb uninstall io.appium.uiautomator2.server.test
